packages/@uppy/aws-s3/src/index.ts

```diff
@@ -385,10 +386,59 @@ export default class AwsS3Multipart<
     return this.#client
   }
 
+  #setClient(opts?: Partial<AwsS3MultipartOptions<M, B>>) {
```

Comment:
I still think this is a lot of code for backwards compatibility which we're not obliged to do. People can not blindly upgrade Uppy anyway, so you have to already check all things of the migration guide anyway. Why not just errors in the constructor, saying the properties are renamed?

Maybe we should let @mifi settle it, backwards compatible or breaking for 4.x.

Reply:
we are breaking many other apis in 4.x, so why not break this too?

Reply:
This does not have any backward compatiblity, it's just emitting warnings if folks are using the old option names.